src/index.ts

```diff
@@ -0,0 +1,1003 @@
+        key = bodyKey(number, hash);
+        value = rlp.encode(body);
+        dbOps.push({
+          type: "put",
+          key: key,
+          keyEncoding: "binary",
+          valueEncoding: "binary",
+          value: value
+        });
+        self.dbManager._cache.body.set(key, value);
+      }
+
+      // if total difficulty is higher than current, add it to canonical chain
+      if (block.isGenesis() || td.gt(currentTd.header)) {
+        self._headHeader = hash;
+        if (!isHeader) {
+          self._headBlock = hash;
+        }
+        if (block.isGenesis()) {
+          self._genesis = hash;
+        }
+
+        // delete higher number assignments and overwrite stale canonical chain
+        async.parallel(
+          [
+            cb => self._deleteStaleAssignments(number.addn(1), hash, dbOps, cb),
+            cb => self._rebuildCanonical(header, dbOps, cb)
+          ],
+          next
+        );
+      } else {
+        if (td.gt(currentTd.block) && !isHeader) {
+          self._headBlock = hash;
+        }
+        // save hash to number lookup info even if rebuild not needed
+        key = hashToNumberKey(hash);
+        value = bufBE8(number);
+        dbOps.push({
+          type: "put",
+          key: key,
+          keyEncoding: "binary",
+          valueEncoding: "binary",
+          value: value
+        });
+        self.dbManager._cache.hashToNumber.set(key, value);
+        next();
+      }
+    }
+  }
+
+  /**
+   *Gets a block by its hash
+   */
+  getBlock(blockTag: Buffer | number | BN, cb: any) {
+    // ensure init completed
+    this._initLock.await(() => {
+      this._getBlock(blockTag, cb);
+    });
+  }
+
+  _getBlock(blockTag: Buffer | number | BN, cb: any) {
+    util.callbackify(this.dbManager.getBlock.bind(this.dbManager))(
+      blockTag,
+      cb
+    );
+  }
+
+  /**
+   * Looks up many blocks relative to blockId
+   */
+  getBlocks(
+    blockId: Buffer | number,
+    maxBlocks: number,
+    skip: number,
+    reverse: boolean,
+    cb: any
+  ) {
+    const self = this;
+    const blocks: any[] = [];
+    let i = -1;
+
+    function nextBlock(blockId: any) {
+      self.getBlock(blockId, function(err?: any, block?: any) {
+        i++;
+
+        if (err) {
+          if (err.notFound) {
+            return cb(null, blocks);
+          } else {
+            return cb(err);
+          }
+        }
+
+        const nextBlockNumber = new BN(block.header.number).addn(
+          reverse ? -1 : 1
+        );
+
+        if (i !== 0 && skip && i % (skip + 1) !== 0) {
+          return nextBlock(nextBlockNumber);
+        }
+
+        blocks.push(block);
+
+        if (blocks.length === maxBlocks) {
+          return cb(null, blocks);
+        }
+
+        nextBlock(nextBlockNumber);
+      });
+    }
+
+    nextBlock(blockId);
+  }
+
+  /**
+   * Gets block details by its hash
+   * @deprecated
+   */
+  getDetails(_: string, cb: any) {
```

Comment:
Is this ``hash`` -> ``_`` replacement some convention to mark things as deprecated? Other reason?

Reply:
The `hash` argument isn't used so it's a convention to note that it's not used